src/main/com/aca/homework/week11/repository/invoice/HashSetInvoiceRepository.java

```diff
@@ -0,0 +1,46 @@
+package com.aca.homework.week11.repository.invoice;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class HashSetInvoiceRepository implements InvoiceRepository {
+
+    private final Set<Invoice> invoices;
+
+    public HashSetInvoiceRepository() {
+        invoices = new HashSet<>();
+    }
+
+    @Override
+    public Invoice save(Invoice invoice) {
+        if (invoice == null)
+            throw new NullInvoiceException();
+        if(invoice.getId() == null)
+            throw new NullIdException();
+        invoices.add(invoice);
+        return invoice;
+    }
+
+    @Override
+    public Invoice findById(String id) {
+        if (id == null)
+            throw new NullIdException();
+        for (Invoice invoice : invoices) {
```

Comment:
because you implemented hashCode and equals for id field only here you can use a small trick. Create a new Invoice with same id and invoke set's contains method